### PR TITLE
fix: Update UI component for improved design

### LINKED ISSUE #243: Bug: Misaligned Buttons in Course Cards on the Homepage Due to Varying Content Length
### Describe the bug

The buttons "Start the Intro to Open Source Course" and "Start the Becoming a Maintainer Course" are misaligned in the card components. The button in the "Becoming a Maintainer" card appears higher than the button in the "Intro to Open Source Course" card due to differences in the content length of the two cards. This creates an inconsistent user experience and disrupts the visual layout of the page.
Before

### Steps to reproduce

1. Navigate to the page that displays both the "Intro to Open Source Course" and "Becoming a Maintainer Course" cards.
2. Observe that the "Becoming a Maintainer" card has less content compared to the "Intro to Open Source Course" card.
3. Notice that the button for "Start the Becoming a Maintainer Course" is positioned higher than the button for "Start the Intro to Open Source Course."
4. The misalignment becomes more evident on varying screen sizes.

### PR DESCRIPTION
**Pull request by Codexnever**
## Description

This PR updates the UI of the card component to improve layout and responsiveness. Additionally, the button within the card is positioned at the bottom for better usability and visual appeal. Please note that the deploy preview link is currently not working, but the changes have been tested locally and are functioning as expected.

## Related Tickets & Documents

Closes #243

## Mobile & Desktop Screenshots/Recordings

**Before**
Before

**After**
After

## Steps to QA
 

1. Pull the latest changes.
2. Run the project locally.
3. Verify that the card layout is responsive and the button is correctly positioned at the bottom.
4. Ensure no issues arise with other components.

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [ ] Tier 4

### CODE REVIEW

**Review by adiati98 — CHANGES_REQUESTED** [review]
Hi @Codexnever, please don't delete sections in the PR template.

You must complete the template, and when a section is not related to your PR, you can leave it blank.

Have you read all the comments in the PR template?
One of it says:

```markdown
<!-- note: PRs with deleted sections will be marked invalid -->
```

That being said, please fix your PR template before we review your PR. Alternatively, you can close this PR and create a new one.

Thank you. 🙂

**Comment by Codexnever** [comment]
> Hi @Codexnever, please don't delete sections in the PR template.
> 
> You must complete the template, and when a section is not related to your PR, you can leave it blank.
> 
> Have you read all the comments in the PR template? One of it says:
> 
> ```
> 
> ```
> 
> That being said, please fix your PR template before we review your PR. Alternatively, you can close this PR and create a new one.
> 
> Thank you. 🙂



Hi @adiati98,

Thank you for your feedback! I apologize for the oversight and for deleting sections of the PR template. I will update the PR template to ensure that all sections are retained and follow the guidelines correctly

Thanks again for your understanding!  🙂

**Comment by adiati98** [comment]
Do let me know once you've fixed it. 👍🏼 
You can take a look at closed PRs to see what you've removed.

**Comment by Codexnever** [comment]
I will update check and let me know if anything remain

**Comment by adiati98** [comment]
@Codexnever your PR should look like this.
 
Screenshot 2024-10-14 165353

**Comment by Codexnever** [comment]
> @Codexnever your PR should look like this.
> 
> Screenshot 2024-10-14 165353


Hey @adiati98 ,

Sorry for submitting the wrong template earlier. Now i corrected you can check

Thanks for understanding!

Best,
Chaitanya

File: package-lock.json
Comment (BekahHW) [inline package-lock.json:1]: Hey @Codexnever can you explain what's going on with the package-lock.json here?

Comment (Codexnever) [reply package-lock.json:1]: Hey @BekahHW ,

The changes in the package-lock.json might have been due to running npm install after making updates. I didn't intend to modify any dependencies directly, so I’ll take a closer look to make sure there’s nothing unexpected.

Let me know if you have any specific concerns!

Comment (Codexnever) [reply package-lock.json:1]: Hi @adiati98 @BekahHW 

Is that any Update ?

Comment (adiati98) [reply package-lock.json:1]: @Codexnever have you taken a closer look to this? To be clear, we don't want to have any changes in this file.

Comment (Codexnever) [reply package-lock.json:1]: @adiati98 Now i make changes in `package-lock-json` as per main branch.

**Review by adiati98 — APPROVED** [review]
Thank you for your contribution, @Codexnever! ✨ 

thank you gif

**Review by BekahHW — APPROVED** [review]

**Pull request merged**